enterprise-suite/templates/prometheus.yaml

```diff
@@ -91,10 +91,12 @@ spec:
     spec:
       serviceAccountName: prometheus-server
 
-      {{ if .Values.podUID }}
       securityContext:
+      {{ if .Values.podUID }}
         runAsUser: {{ .Values.podUID }}
       {{ end }}
+        # Needed so the prometheus data volume is write-able. The official prometheus image runs with gid 65534.
+        fsGroup: 65534
```

Comment:
@jsravn I'm curious how you discovered what group the prometheus image runs as?

Reply:
`kubectl exec <pod> -- groups` -> shows `nogroup`
`kubectl exec <pod> -- grep nogroup /etc/groups` -> shows 65534

I'd rather the prometheus pod ran as the root user though - I think the usage of non-root is going to cause complications with openshift (yet to be tested).

Reply:
Cool.  I understand running as root is the easiest most cross-platform compatible way, but I also understand the gaping security hole it opens up.  I haven't spent a lot of time understanding how to secure container deployments, but simply running as nobody, or worse, some random user/UID is obviously going to break depending on your environment.

I like the user namespace feature in Docker (https://docs.docker.com/engine/security/userns-remap/), though I've only read about it.  I'm not sure how or if it's that portable across K8s deploys since it relies on docker daemon config.

Reply:
Also, how did you troubleshoot the prometheus image by running commands in the pod?  Did you run those commands on an environment that worked (i.e. minikube), or did you do something to keep the container running long enough to execute it on GKE?

Reply:
@seglo Openshift automatically maps the container user to an isolated UID (1000+), so running root containers in Openshift is fine.

In other environments it's not ideal - but there isn't really a great way to handle this - since the container has no way of knowing how that UID is used on the host system, and uids are not namespaced on Linux systems. It would be really nice if Openshift upstreamed this functionality.

The userns remapping is a possibility but not natively supported in K8s yet as far as I know. What you can do though is set `runAsUser` in the security context for a container - which is a good middle ground (but again, it depends on the host system's UIDs, so in a helm chart we can't default that to anything meaingful).

Reply:
> Also, how did you troubleshoot the prometheus image by running commands in the pod? Did you run those commands on an environment that worked (i.e. minikube), or did you do something to keep the container running long enough to execute it on GKE?

I did a `kubectl edit deployment` on `prometheus-server`, and attached the prometheus mount to inspect how it was getting mounted. To inspect the prometheus container, I just looked at my local minikube - but I think you could as easily do a `docker run` and inspect it that way.

Reply:
Thanks @jsravn !